myFilter.cpp

```diff
@@ -0,0 +1,286 @@
+#include "myFilter.hpp"
+#include "opencv2/imgproc/imgproc.hpp"
+#include "math.h"
+#include <stack>
+#include <iostream>
+#include "highgui.h"
+
+void myCanny(const cv::Mat & grayscale, cv::Mat edges, int threshold1, int threshold2)
+{
+    cv::Mat gradX, gradY;
+    cv::Mat gradMagn(grayscale.rows, grayscale.cols, CV_32FC1, 0.0);
+    cv::Mat gradAngle(grayscale.rows, grayscale.cols, CV_32FC1, 0.0);
+
+    cv::Sobel(grayscale, gradX, CV_32FC1, 1, 0);
+    cv::Sobel(grayscale, gradY, CV_32FC1, 0, 1);
+
+    float PI = atan(1)*4;
+
+    for(int i = 0; i < gradX.rows; i++)
+    {
+        for(int j = 0; j < gradX.cols; j++)
+        {
+            gradMagn.at<float>(i,j) = sqrt((gradX.at<float>(i,j)*gradX.at<float>(i,j)) + (gradY.at<float>(i,j)*gradY.at<float>(i,j)));
```

Comment:
it's not optimal to use at<> operator. I suggest using ptr style:
gradMagn_row_ptr = gradMagn.ptr<float>(r);
for(int c = 0; c < cols; c++) { do_smth(gradMagn_row_ptr[c]); }